lightning/src/util/dyn_signer.rs

```diff
@@ -0,0 +1,459 @@
+    }
+
+    fn sign_holder_htlc_transaction(&self, htlc_tx: &Transaction, input: usize, htlc_descriptor: &HTLCDescriptor, secp_ctx: &Secp256k1<All>) -> Result<secp256k1::schnorr::Signature, ()> {
+        todo!();
+    }
+
+    fn sign_counterparty_htlc_transaction(&self, htlc_tx: &Transaction, input: usize, amount: u64, per_commitment_point: &PublicKey, htlc: &HTLCOutputInCommitment, secp_ctx: &Secp256k1<All>) -> Result<secp256k1::schnorr::Signature, ()> {
+        todo!();
+    }
+
+    fn partially_sign_closing_transaction(&self, closing_tx: &ClosingTransaction, secp_ctx: &Secp256k1<All>) -> Result<PartialSignature, ()> {
+        todo!();
+    }
+
+    fn sign_holder_anchor_input(&self, anchor_tx: &Transaction, input: usize, secp_ctx: &Secp256k1<All>) -> Result<secp256k1::schnorr::Signature, ()> {
+        todo!();
+    }
+}
+
+impl Clone for DynSigner {
+    fn clone(&self) -> Self {
+        DynSigner { inner: self.inner.box_clone() }
+    }
+}
+
+// This is taken care of by KeysInterface
+impl Readable for DynSigner {
+    fn read<R: Read>(_reader: &mut R) -> Result<Self, DecodeError> {
+        unimplemented!()
+    }
+}
+
+impl EcdsaChannelSigner for DynSigner {
+    fn sign_holder_commitment(
+        &self, commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<secp256k1::All>,
+    ) -> Result<Signature, ()> {
+        self.inner.sign_holder_commitment(commitment_tx, secp_ctx)
+    }
+
+    #[cfg(any(test, feature = "unsafe_revoked_tx_signing"))]
+    fn unsafe_sign_holder_commitment(
+        &self, commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<secp256k1::All>,
+    ) -> Result<Signature, ()> {
+        self.inner.unsafe_sign_holder_commitment(commitment_tx, secp_ctx)
+    }
+
+    fn sign_counterparty_commitment(
+        &self, commitment_tx: &CommitmentTransaction, inbound_htlc_preimages: Vec<PaymentPreimage>,
+        outbound_htlc_preimages: Vec<PaymentPreimage>, secp_ctx: &Secp256k1<secp256k1::All>,
+    ) -> Result<(Signature, Vec<Signature>), ()> {
+        self.inner.sign_counterparty_commitment(commitment_tx, inbound_htlc_preimages, outbound_htlc_preimages, secp_ctx)
+    }
+
+    fn sign_justice_revoked_output(
+        &self,
+        justice_tx: &Transaction,
+        input: usize,
+        amount: u64,
+        per_commitment_key: &SecretKey,
+        secp_ctx: &Secp256k1<secp256k1::All>,
+    ) -> Result<Signature, ()> {
+        EcdsaChannelSigner::sign_justice_revoked_output(&*self.inner, justice_tx, input, amount, per_commitment_key, secp_ctx)
+    }
+
+    fn sign_justice_revoked_htlc(
+        &self,
+        justice_tx: &Transaction,
+        input: usize,
+        amount: u64,
+        per_commitment_key: &SecretKey,
+        htlc: &HTLCOutputInCommitment,
+        secp_ctx: &Secp256k1<secp256k1::All>,
+    ) -> Result<Signature, ()>  {
+        EcdsaChannelSigner::sign_justice_revoked_htlc(&*self.inner, justice_tx, input, amount, per_commitment_key, htlc, secp_ctx)
```

Comment:
name clashes are inconvenient, result in this need to disambiguate, and can't use `delegate!` macro